testing/tests/test_e2e/test_event.py

```diff
@@ -90,9 +90,10 @@ def _foo(self, e):
         ctx.on.start(), State(deferred=[_Event("update-status").deferred(MyCharm._foo)])
     )
 
-    assert len(ctx.emitted_events) == 5
```

Comment:
Right after this we check for the exact contents of the list, so there doesn't seem to be any point checking the length first.